### PR TITLE
Add support for using sni to enabled/disable 0-rtt

### PR DESCRIPTION
**Pull request by duke8253**

### CODE REVIEW

File: iocore/net/TLSSNISupport.cc [hunk 1]
```diff
@@ -130,8 +130,56 @@ TLSSNISupport::on_servername(SSL *ssl, int *al, void *arg)
   this->_fire_ssl_servername_event();
 
   const char *name = SSL_get_servername(ssl, TLSEXT_NAMETYPE_host_name);
+
   if (name) {
     this->_set_sni_server_name(name);
+
+#if TS_HAS_TLS_EARLY_DATA
```
Comment (maskit) [inline iocore/net/TLSSNISupport.cc:137]: I don't want to have code for early data here. One or two lines maybe acceptable if it's really needed. And DRY.

**Comment by maskit** [comment]
Why don't you use `hints_from_sni`? You don't have to have `server_early_data_map` and look up the table by server name again if you set the config value in an SNI action.

File: tests/gold_tests/tls/tls_0rtt_server.test.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 '''
 '''
 #  Licensed to the Apache Software Foundation (ASF) under one
-#  or more contributor license agreements.  See the NOTICE file
+#  or more contributor license agreements1.  See the NOTICE file
```
Comment (maskit) [inline tests/gold_tests/tls/tls_0rtt_server.test.py:4]: 1?

File: iocore/net/SSLNetVConnection.cc [hunk 3]
```diff
@@ -1298,6 +1259,51 @@ SSLNetVConnection::sslServerHandShakeEvent(int &err)
     } // Still data in the BIO
   }
 
+#if TS_HAS_TLS_EARLY_DATA
```
Comment (maskit) [inline iocore/net/SSLNetVConnection.cc:1262]: Just to make sure, this code is moved to here to run SNI actions first, right?

**Review by maskit — COMMENTED** [review]
Looks good. Waiting for CI jobs complete.

**Comment by duke8253** [comment]
I think this is ready for review now.

**Comment by duke8253** [comment]
[approve ci]

**Comment by duke8253** [comment]
[approve ci debian]

**Comment by duke8253** [comment]
[approve ci debian]

**Review by maskit — APPROVED** [review]
Just curious, but was there any issue with adding a function to give NetVC a chance to check hints_from_sni in a common way?

**Pull request merged**